sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPlugin.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.ServiceBus.Plugins
+{
+    /// <summary>
+    /// This class provides methods that can be overridden to manipulate messages for custom plugin functionality.
+    /// </summary>
+    public abstract class ServiceBusPlugin
+    {
+        /// <summary>
+        /// This operation is called before a message is sent and can be
+        /// overridden to alter the properties of an outgoing message.
+        /// </summary>
+        /// <param name="message">The <see cref="ServiceBusMessage"/> to be modified by the plugin.</param>
+        public virtual Task BeforeMessageSend(ServiceBusMessage message)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// This operation is called after a message is received, but before it is returned to the <see cref="ServiceBusReceiver"/>.
+        /// It can be overriden to alter the properties of an
```

Comment:
Whoever is reading the message might interpret that it's only possible to alter the message properties but not the body.

```suggestion
        /// It can be overridden to alter the body and the properties of an
```